src/App.tsx

```diff
@@ -17,18 +75,38 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                filterBy={filterBy}
+                setFilterBy={setFilterBy}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!error
+                ? (
+                  <TodoList
+                    todos={visibleTodos}
+                    openTodo={setSelectedTodo}
+                    selectedTodo={selectedTodo}
+                  />
+                ) : (
+                  <span>Error while loading data from server</span>
+                )}
+
+              {isLoading && <Loader />}
```

Comment:
With this logic, I see a loader and an empty table.
Show TodoList only when the `isLoading` is false.